Build a module with periods in the module name

**Description:**
When building a module consisting of periods (like full-stop), a bad-sad error is thrown.

```
[2020-02-05 17:12:40,450] SEVERE {b7a.log.crash} - java.lang.NullPointerException 
org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
	at org.ballerinalang.testerina.core.BTestRunner.lambda$processProgramFile$4(BTestRunner.java:316)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.ballerinalang.testerina.core.BTestRunner.processProgramFile(BTestRunner.java:308)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$buildSuites$2(BTestRunner.java:285)
	at java.util.HashMap.forEach(HashMap.java:1289)
	at org.ballerinalang.testerina.core.BTestRunner.buildSuites(BTestRunner.java:273)
	at org.ballerinalang.testerina.core.BTestRunner.runTest(BTestRunner.java:156)
	at org.ballerinalang.testerina.util.TesterinaUtils.executeTests(TesterinaUtils.java:132)
	at org.ballerinalang.packerina.task.RunTestsTask.execute(RunTestsTask.java:79)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:394)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:59)
Caused by: java.lang.NullPointerException
	at sun.net.util.URLUtil.urlNoFragString(URLUtil.java:50)
	at sun.misc.URLClassPath.getLoader(URLClassPath.java:522)
	at sun.misc.URLClassPath.getNextLoader(URLClassPath.java:494)
	at sun.misc.URLClassPath.getResource(URLClassPath.java:248)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.ballerinalang.testerina.util.TestarinaClassLoader.loadClass(TestarinaClassLoader.java:60)
	at org.ballerinalang.testerina.core.BTestRunner.packageProcessed(BTestRunner.java:366)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$processProgramFile$4(BTestRunner.java:311)
```

**Steps to reproduce:**
Create a new ballerina project
Add a module name with period (eg: foo.bar)
Build the module

The stacktrace is as follows:

**Suggested Labels (optional):**
Area/BuildTools, Type/Bug

--- a/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java
+++ b/misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java
@@ -64,6 +64,7 @@ import java.util.stream.Collectors;
 public class BTestRunner {
 
     public static final String MODULE_INIT_CLASS_NAME = "___init";
+    private static final String FILE_NAME_PERIOD_SEPARATOR = "$$$";
 
     private PrintStream errStream;
     private PrintStream outStream;
@@ -252,9 +253,8 @@ public class BTestRunner {
         boolean hasTestablePackage = !packageName.equals(TesterinaConstants.DOT);
         if (hasTestablePackage) {
             // Load test init class
-            String testClassName = TesterinaUtils.getQualifiedClassName(suite.getOrgName(),
-                                                                        suite.getPackageID(),
-                                                                        suite.getPackageID());
+            String testClassName = TesterinaUtils.getQualifiedClassName(suite.getOrgName(), suite.getPackageID(),
+                    suite.getPackageID().replace(".", FILE_NAME_PERIOD_SEPARATOR));
             try {
                 testInitClazz = classLoader.loadClass(testClassName);
             } catch (Throwable e) {

--- a/tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingTestCase.java
+++ b/tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingTestCase.java
@@ -253,6 +253,39 @@ public class PackagingTestCase extends BaseTest {
                 new LogLeecher[]{clientLeecher}, tempProjectDirectory.toString());
     }
 
+    @Test(description = "Test and run a module which has a module name contains period. eg: foo.bar")
+    public void testBuildAndRunModuleWithPeriod() throws BallerinaTestException {
+        // Test ballerina init
+        Path projectPath = tempProjectDirectory.resolve("buildAndRunModuleWithPeriodProject");
+
+        // Create project
+        balClient.runMain("new", new String[] { "buildAndRunModuleWithPeriodProject" }, envVariables, new String[] {},
+                new LogLeecher[] {}, projectPath.getParent().toString());
+
+        Assert.assertTrue(Files.exists(projectPath));
+        Assert.assertTrue(Files.isDirectory(projectPath));
+
+        // Create module named `foo.bar`
+        String moduleName = "foo.bar";
+        balClient.runMain("add", new String[] { moduleName }, envVariables, new String[] {}, new LogLeecher[] {},
+                projectPath.toString());
+
+        Assert.assertTrue(Files.exists(projectPath.resolve("src").resolve(moduleName)));
+        Assert.assertTrue(Files.isDirectory(projectPath.resolve("src").resolve(moduleName)));
+
+        // Build module
+        LogLeecher buildLeecher = new LogLeecher("[pass] testFunction");
+        balClient.runMain("build", new String[] { "-c", "-a" }, envVariables, new String[] {},
+                new LogLeecher[] { buildLeecher }, projectPath.toString());
+        buildLeecher.waitForText(60000);
+
+        // Run module
+        LogLeecher runLeecher = new LogLeecher("Hello World!");
+        balClient.runMain("run", new String[] { moduleName }, envVariables, new String[] {},
+                new LogLeecher[] { runLeecher }, projectPath.toString());
+        buildLeecher.waitForText(5000);
+    }
+
     /**
      * Get environment variables and add ballerina_home as a env variable the tmp directory.
      *

